atlasdb-coordination-impl/src/main/java/com/palantir/atlasdb/coordination/keyvalue/KeyValueServiceCoordinationStore.java

```diff
@@ -125,18 +125,45 @@ private KeyValueServiceCoordinationStore(
     @Override
     public CheckAndSetResult<ValueAndBound<T>> transformAgreedValue(Function<ValueAndBound<T>, T> transform) {
         Optional<SequenceAndBound> coordinationValue = getCoordinationValue();
-        T targetValue = transform.apply(
-                ValueAndBound.of(coordinationValue.flatMap(
-                        sequenceAndBound -> getValue(sequenceAndBound.sequence())),
-                        coordinationValue.map(SequenceAndBound::bound).orElse(SequenceAndBound.INVALID_BOUND)));
+        ValueAndBound<T> extantValueAndBound = ValueAndBound.of(coordinationValue.flatMap(
```

Comment:
This new logic looks like it should be fine. Whether we reuse a value or not, we still advance the bound, and the check and set request will still conflict appropriately since it includes the bound. Values are immutable if, so no issues there with reuse.